### ISSUE TITLE
Fresh install of Knet on julia 0.5.2 gives wrong answer  on all examples

### ISSUE THREAD

**Issue by ngphuoc** [denizyuret/Knet.jl#127]
I have updated my julia to 0.5.2 and Knet 0.8.3 (remove ~/.julia and fresh Pkg.add("Knet")). I tried the following example but it gave incorrect results. Should I stay with 0.5.1?:
```julia
$ julia housing.jl 
housing.jl (c) Deniz Yuret, 2016. Linear regression model for the Housing dataset from the UCI Machine Learning
Repository.
opts=(:seed,-1)(:epochs,20)(:lr,0.1)(:atype,"KnetArray{Float32}")(:test,0.0)(:gcheck,0)(:fast,false)
size(data) = (14,506)
(:epoch,0,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,1,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,2,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,3,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,4,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,5,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,6,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,7,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,8,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,9,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,10,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,11,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,12,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,13,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,14,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,15,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,16,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,17,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,18,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,19,:trn,-4.2498277f18,:tst,-4.2498277f18)
(:epoch,20,:trn,-4.2498277f18,:tst,-4.2498277f18)
  1.097184 seconds (1.06 M allocations: 46.389 MB, 0.92% gc time)
```

**Comment by ngphuoc** [comment]
It also failed on julia 0.5.1:

```julia
~/.julia/v0.5/Knet/examples >julia housing.jl 
housing.jl (c) Deniz Yuret, 2016. Linear regression model for the Housing dataset from the UCI Machine Learning
Repository.
opts=(:seed,-1)(:epochs,20)(:lr,0.1)(:atype,"KnetArray{Float32}")(:test,0.0)(:gcheck,0)(:fast,false)
size(data) = (14,506)
(:epoch,0,:trn,0.0f0,:tst,0.0f0)
(:epoch,1,:trn,0.0f0,:tst,0.0f0)
(:epoch,2,:trn,0.0f0,:tst,0.0f0)
(:epoch,3,:trn,0.0f0,:tst,0.0f0)
(:epoch,4,:trn,0.0f0,:tst,0.0f0)
(:epoch,5,:trn,0.0f0,:tst,0.0f0)
(:epoch,6,:trn,0.0f0,:tst,0.0f0)
(:epoch,7,:trn,0.0f0,:tst,0.0f0)
(:epoch,8,:trn,0.0f0,:tst,0.0f0)
(:epoch,9,:trn,0.0f0,:tst,0.0f0)
(:epoch,10,:trn,0.0f0,:tst,0.0f0)
(:epoch,11,:trn,0.0f0,:tst,0.0f0)
(:epoch,12,:trn,0.0f0,:tst,0.0f0)
(:epoch,13,:trn,0.0f0,:tst,0.0f0)
(:epoch,14,:trn,0.0f0,:tst,0.0f0)
(:epoch,15,:trn,0.0f0,:tst,0.0f0)
(:epoch,16,:trn,0.0f0,:tst,0.0f0)
(:epoch,17,:trn,0.0f0,:tst,0.0f0)
(:epoch,18,:trn,0.0f0,:tst,0.0f0)
(:epoch,19,:trn,0.0f0,:tst,0.0f0)
(:epoch,20,:trn,0.0f0,:tst,0.0f0)
  1.310770 seconds (1.08 M allocations: 47.610 MB, 0.94% gc time)
```

**Comment by AStupidBear** [comment]
@ngphuoc @denizyuret  I think we have the same problem #125.

**Comment by denizyuret** [comment]
What is the GPU chip and library/driver versions you are using?  Does a+b
work for KnetArrays?

**Comment by denizyuret** [comment]
OK, the solution that resolved #125 may work for you as well: please remove "-arch=compute_30 -code=sm_30" from Makefile and run "make clean; make" again in Knet/src.

**Comment by ngphuoc** [comment]
Mine is GeForce GTX TITAN X, Driver Version: 375.26. 

$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2016 NVIDIA Corporation
Built on Tue_Jan_10_13:22:03_CST_2017
Cuda compilation tools, release 8.0, V8.0.61

Yes that solution works for me! Many thanks.
